src/components/ValueSelectorCard/ValueSelectorCard.js

```diff
@@ -0,0 +1,338 @@
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import * as React from 'react';
+import styled from 'styled-components/native';
+import t from 'tcomb-form-native';
+import isEmpty from 'lodash.isempty';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import get from 'lodash.get';
+
+import ShadowedCard from 'components/ShadowedCard';
+import { BaseText } from 'components/Typography';
+
+import type { Assets, Balances, Rates } from 'models/Asset';
+import type { RootReducerState } from 'reducers/rootReducer';
+import type { TransactionFeeInfo } from 'models/Transaction';
+import type { FormSelector } from 'models/TextInput';
+
+import { formatAmount, formatFiat } from 'utils/common';
+import { spacing } from 'utils/variables';
+import { getBalance, getRate, sortAssets, calculateMaxAmount } from 'utils/assets';
+import { themedColors } from 'utils/themes';
+import { SelectorInputTemplate, selectorStructure, inputFormatter, inputParser } from 'utils/formHelpers';
+import { getFormattedBalanceInFiat } from 'screens/Exchange/utils';
+
+import { accountBalancesSelector } from 'selectors/balances';
+import { accountAssetsSelector } from 'selectors/assets';
+import { defaultFiatCurrency } from 'constants/assetsConstants';
+
+
+type Props = {
+  assets: Assets,
+  balances: Balances,
+  baseFiatCurrency: ?string,
+  maxLabel?: string,
+  preselectedAsset?: string,
+  rates: Rates,
+  getFormValue: (?FormSelector) => void,
+  txFeeInfo?: ?TransactionFeeInfo,
+};
+
+type FormValue = {
+  formSelector: FormSelector,
+};
+
+type State = {
+  formOptions: Object,
+  value: FormValue,
+  errorMessage: string,
+};
+
+const Wrapper = styled.View`
+  width: 100%;
+  padding: ${spacing.layoutSides}px;
+`;
+
+const FormWrapper = styled.View`
+  width: 100%;
+`;
+
+const ErrorMessage = styled(BaseText)`
+  margin: ${spacing.medium}px 0;
+  color: ${themedColors.negative};
+  text-align: center;
+`;
+
+const { Form } = t.form;
+
+const getFormStructure = (balances: Balances, txFeeInfo: ?TransactionFeeInfo) => {
+  return t.struct({
+    formSelector: selectorStructure(balances, false, txFeeInfo),
+  });
+};
+
+export class ValueSelectorCard extends React.Component<Props, State> {
+  form: Form;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      value: {
+        formSelector: {
+          selector: {},
+          input: '',
+        },
+      },
+      formOptions: {
+        fields: {
+          formSelector: {
+            keyboardType: 'decimal-pad',
```

Comment:
Can we please have an option prop to fill only integers without the decimal option ?